lsp/src/lib.rs

```diff
@@ -41,36 +42,31 @@ impl Backend {
     }
 
     let start = Instant::now();
-    let mut diags = match lsp.compiler.compile(()) {
-      Ok(_) => Vec::new(),
-      Err(diags) => diags,
-    };
+    _ = lsp.compiler.compile(());
     eprintln!("compiled in {:?}", start.elapsed());
 
-    diags.sort_by_key(|d| Some(d.span()?.file));
-    let mut diags = diags.into_iter().peekable();
-    while diags.peek().is_some_and(|x| x.span().is_none()) {
-      diags.next();
+    let mut diags_by_file = HashMap::<FileId, (Vec<&Diag>, Vec<&Diag>)>::new();
```

Comment:
should we maybe name `(Vec<&Diag>, Vec<&Diag>)` so we know which are errors and which are warnings?